mava/wrappers/matrax.py

```diff
@@ -21,30 +21,33 @@
 from jumanji.types import TimeStep
 from jumanji.wrappers import Wrapper
 
-from mava.types import Observation, State
+from mava.types import Observation, ObservationGlobalState, State
 
 
 class MatraxWrapper(Wrapper):
     """Multi-agent wrapper for the Matrax environment."""
 
-    def __init__(self, env: Environment):
+    def __init__(self, env: Environment, add_global_state: bool):
         super().__init__(env)
         self._num_agents = self._env.num_agents
         self._num_actions = self._env.num_actions
         self.action_mask = jnp.ones((self._num_agents, self._num_actions), dtype=bool)
+        self.add_global_state = add_global_state
 
-    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+    def modify_timestep(
+        self, timestep: TimeStep
+    ) -> TimeStep[Union[Observation, ObservationGlobalState]]:
         """Modify the timestep for `step` and `reset`."""
-        observation = Observation(
-            agents_view=timestep.observation.agent_obs,
-            action_mask=self.action_mask,
-            step_count=jnp.repeat(timestep.observation.step_count, self._num_agents),
-        )
-        return timestep.replace(
-            observation=observation,
-            reward=timestep.reward,
-            discount=timestep.discount,
-        )
+        obs_data = {
+            "agents_view": timestep.observation.agent_obs,
+            "action_mask": self.action_mask,
+            "step_count": jnp.repeat(timestep.observation.step_count, self._num_agents),
+        }
+        if self.add_global_state:
+            obs_data["global_state"] = timestep.observation.agent_obs
```

Comment:
Even though the obs are just zeros, I wonder if we shouldn't still concatenate here just to make the centralised critic condition on something of at least a different shape to the agents.

Reply:
Okay. Its changed now.